### PR TITLE
[FIX] Rename Omnichannel Rooms, Inquiries and Subscriptions when the Contact Name changes

### PR DESCRIPTION
**Pull request by rafaelblink**
### Before
https://user-images.githubusercontent.com/2493803/114251922-37220100-9979-11eb-83eb-23ce51645192.mp4

### After
https://user-images.githubusercontent.com/2493803/114251546-ccbc9100-9977-11eb-9699-2fd3c054a9c1.mp4

### CODE REVIEW

File: app/livechat/server/api/v1/contact.js [hunk 1]
```diff
@@ -21,9 +21,9 @@ API.v1.addRoute('omnichannel/contact', { authRequired: true }, {
 				contactManager: Match.Maybe(Object),
 			});
 
-			const contact = Contacts.registerContact(this.bodyParams);
+			const contactId = Contacts.registerContact(this.bodyParams);
 
-			return API.v1.success({ contact });
+			return API.v1.success({ contact: contactId });
```
Comment (renatobecker) [inline app/livechat/server/api/v1/contact.js:26]: Why did you change the original code if the result is the same?
I didn't get it.

File: app/livechat/server/lib/Contacts.js [hunk 2]
```diff
@@ -60,6 +64,15 @@ export const Contacts = {
 
 		LivechatVisitors.updateById(contactId, updateUser);
 
+		const room = LivechatRooms.findOneByVisitorToken(token, { _id: 1 });
```
Comment (renatobecker) [inline app/livechat/server/lib/Contacts.js:67]: This doesn't make sense, this method is returning a random room, you can't make sure this is the room that needs to be updated in case the contact has more than just one room stored on the database.

My Suggestion: 
- Fetch all rooms associated with the same contact and update all of them.

**Review by renatobecker — CHANGES_REQUESTED** [review]

File: app/livechat/server/lib/Contacts.js [hunk 3]
```diff
@@ -64,13 +64,15 @@ export const Contacts = {
...
-
-		if (room) {
-			const { _id: rid } = room;
-			return Rooms.setFnameById(rid, name)
-				&& LivechatInquiry.setNameByRoomId(rid, name)
-				&& Subscriptions.updateDisplayNameByRoomId(rid, name);
+		const rooms = LivechatRooms.findByVisitorId(contactId).fetch();
+
+		if (rooms && rooms.length > 0) {
+			rooms.forEach((room) => {
```
Comment (renatobecker) [inline app/livechat/server/lib/Contacts.js:70]:
```suggestion
			rooms?.length && rooms.forEach((room) => {
```

Comment (renatobecker) [reply app/livechat/server/lib/Contacts.js:70]: @rafaelblink what about this?

**Review by renatobecker — CHANGES_REQUESTED** [review]
Everything looks good, except the `Contact Name` on the Room header.
That information is not being updated in real-time.

Screen Shot 2021-04-20 at 11 21 13 AM

**Comment by rafaelblink** [comment]
> Everything looks good, except the `Contact Name` on the Room header.
> That information is not being updated in real-time.
> 
> Screen Shot 2021-04-20 at 11 21 13 AM

https://user-images.githubusercontent.com/2493803/114251546-ccbc9100-9977-11eb-9699-2fd3c054a9c1.mp4

I don't know which steps you did to not being updated but in this video is updated successfuly.

**Review by renatobecker — APPROVED** [review]
